subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -1442,7 +1459,13 @@ static void isoal_source_deallocate(isoal_source_handle_t hdl)
 	struct isoal_pdu_production *pp;
 	struct isoal_source *source;
 
-	source = &isoal_global.source_state[hdl];
+	if (hdl < ARRAY_SIZE(isoal_global.source_state)) {
+		source = &isoal_global.source_state[hdl];
+	} else {
+		LL_ASSERT(0);
+		return;
```

Comment:
```suggestion
```

Reply:
If we do not return here, the compiler will complain that `source` is uninitialized on line 1469. 

The alternative would be to move the entire part into the `if` statement. What do you prefer?

Reply:
Understood now, we keep as-is.